passwerk_TMSP/passwerk_TMSP.go

```diff
@@ -171,13 +164,18 @@ func (app *PasswerkApplication) CheckTx(tx []byte) types.Result {
 
 	switch operationalOption {
 	case "writing":
-		if len(parts) < 6 {
+		if len(parts) < 7 {
 			return badReturn("Invalid number of TX parts")
 		}
 	case "deleting":
-		if len(parts) < 3 {
+		if len(parts) < 7 { //note that the length for deleting and writing just happens to be the same, may change in future passwerk
 			return badReturn("Invalid number of TX parts")
 		}
+
+		//check to make sure that system state hasn't changed
+		if parts[3] != bytes2HexString(app.state.Hash()) { //here parts[3] has passed on the system state from the broadcast

```

Comment:
Thanks for the catch, parts[3] should be updated to parts[6], which currently doesn't exist, this needs to pass the system state at the time that the tx was submitted. The idea behind checking the state here is to ensure that if multiple txs were submitted simultaneously and they were referencing the same user account it would cause issues under tree_DeleteRecord func, particularly with the map_Index2Remove input value, for instance if two deletions were submitted, one of them may end up deleting the next record in the map. I think that I'm going to restructure the code to prevent the above scenario either way (work around to not have to use map_Index2Remove). This is eliminate the need to check the system state in the code you have pointed out, instead at this point passwerk will check for deleted records existence